### ISSUE TITLE
Why all checks have passed?

### ISSUE THREAD

**Issue by pypygeek** [cpp-linter/cpp-linter-action#117]
hello,

Why all checks have passed?

I want to raise an error if the ansi rule is not met.

```yml
name: C Code Convention CI

on: [push]

jobs:
  cpp-linter:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: cpp-linter/cpp-linter-action@v2
        id: linter
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          style: file
          extensions: c,C,h,H
          extra-args: -Wall -Wextra -ansi -c

      - name: Fail fast?!
        if: steps.linter.outputs.checks-failed > 0
        run: echo "Some files failed the linting checks!"
```

![image](https://user-images.githubusercontent.com/19469996/198834118-d00a3f5c-592f-45c5-90b7-17bd4c01df1a.png)
[Create a.c](https://github.com/pypygeek/action_test/pull/1/files)

**Comment by 2bndy5** [comment]
You need to change the "fail fast" step in your workflow.
```diff
 - name: Fail fast?!
         if: steps.linter.outputs.checks-failed > 0
-        run: echo "Some files failed the linting checks!"
+        run: exit 1
```

**Comment by 2bndy5** [comment]
>  I want to raise an error if the ansi rule is not met.

If you only want to raise an error for a specific rule (and not for other rules), then this might be done in your workflow by running cpp-linter twice -- once for checks that don't raise an error, and once for the ANSI checks that should raise an error. This can be done as 2 separate jobs in the same workflow.

But this is not a problem with the cpp-linter-action because there is no support in clang-tidy to give a higher "rank" to certain checks and a lower "rank" to others.

**Comment by pypygeek** [comment]
> You need to change the "Fail fast?!" step in your workflow.
> 
> ```diff
>        - name: Fail fast?!
>          if: steps.linter.outputs.checks-failed > 0
> -        run: echo "Some files failed the linting checks!"
> +        run: exit 1
> ```

Thank you for your valuable time.

Close the issue by solving the problem.
